packages/react-dom/src/client/ReactDOMFiberTextarea.js

```diff
@@ -126,7 +127,7 @@ export function updateWrapper(element: Element, props: Object) {
   if (value != null) {
     // Cast `value` to a string to ensure the value is set correctly. While
     // browsers typically do this as necessary, jsdom doesn't.
-    const newValue = '' + value;
+    const newValue = getSafeValue(value);
 
     // To avoid side effects (such as losing text selection), only set value if changed
     if (newValue !== node.value) {
```

Comment:
Ah right. We need the new value to be a string for comparison.

Reply:
That does make sense to me. Is there a particular reason though why the casting is done by concatenation instead of e.g. `String(value)`?

Reply:
Maybe we could add a similar check like in [ReactDOMFiberInput](https://github.com/facebook/react/blob/82c7ca4cca90976cd1230e03ea1a4372a4276e67/packages/react-dom/src/client/ReactDOMFiberInput.js#L182)?

Reply:
> because the value always reports as a string

I was wondering why Flow did not catch that and found the issue in the `getSafeValue()` function. Not sure if we want the fix since it adds some complexity but it's nice to know that all code paths indeed work. Check out: https://github.com/facebook/react/pull/13367